### PR TITLE
GH-928: ZooKeeperServer with an auto deletable data directory

### PR DESCRIPTION
**Pull request by apanasevich**
Hi,

here is a fix for [Issue 928](https://github.com/spring-projects/spring-kafka/issues/928). I've created a class to start a ZooKeeper with an auto deletable data dicrectory.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@apanasevich Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-kafka&pullRequestId=932)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-kafka&pullRequestId=932) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@apanasevich Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-kafka&pullRequestId=932)!

**Comment by apanasevich** [comment]
Shall I fix codestyle issues manually or there is a codestyle.xml file for IDEA?

File: spring-kafka-test/src/main/java/org/springframework/kafka/test/ZooKeeperServer.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import org.apache.zookeeper.server.ServerConfig;
+import org.apache.zookeeper.server.ZooKeeperServerMain;
+import org.apache.zookeeper.server.quorum.QuorumPeerConfig;
+
+/**
+ * A zookeeper server to use with unit tests.
+ *
+ * @author Dmitrii Apanasevich
+ */
+public final class ZooKeeperServer {
```
Comment (artembilan) [inline spring-kafka-test/src/main/java/org/springframework/kafka/test/ZooKeeperServer.java]: We can't accept such a breaking change. On the other hand it looks like in the end you are pursuing a `cnxnFactory.shutdown()`, which, looks like is accessed via:
```
this.zookeeper.zookeeper().shutdown();
this.zookeeper.zookeeper().getServerCnxnFactory().shutdown();
```
in the `EmbeddedKafkaBroker`.

I agree about other polishing around magic numbers and missed logs for swallowed exception, but I don't see reason in this class overhead which may lag with updates in the future Apache Kafka versions.

So, WDYT about my suggestion?

Comment (apanasevich) [reply spring-kafka-test/src/main/java/org/springframework/kafka/test/ZooKeeperServer.java]: I have no other options, so I've created [an issue](https://issues.apache.org/jira/browse/ZOOKEEPER-3248) in Apache tracker.

**Comment by artembilan** [comment]
Looks like the issue with regular Kafka Consumer and Produce usage has disappeared with the latest Kafka Client.

We still an issue in Kafka Streams tests, but that's going to be addressed on Apache Kafka side: https://issues.apache.org/jira/browse/KAFKA-6647

So, closed as *Won't Fix*.

Thank you for the effort anyway!

**Pull request closed without merging**